test/byContract/ENSIntegration/ENSIntegration.test.ts

```diff
@@ -0,0 +1,126 @@
+                expect(await resolve('rlc.iexec.eth')).to.equal(await iexecPoco.token());
+            }
+            expect(await resolve('admin.iexec.eth')).to.equal(iexecAdmin.address);
+            expect(await resolve('core.v5.iexec.eth')).to.equal(proxyAddress);
+            expect(await resolve('apps.v5.iexec.eth')).to.equal(await iexecPoco.appregistry());
+            expect(await resolve('datasets.v5.iexec.eth')).to.equal(
+                await iexecPoco.datasetregistry(),
+            );
+            expect(await resolve('workerpools.v5.iexec.eth')).to.equal(
+                await iexecPoco.workerpoolregistry(),
+            );
+        });
+    });
+
+    describe('Reverse resolution', () => {
+        it('Should lookup initial addresses', async () => {
+            expect(await lookup(iexecAdmin.address)).to.equal('admin.iexec.eth');
+            expect(await lookup(proxyAddress)).to.equal('core.v5.iexec.eth');
+            expect(await lookup(await iexecPoco.appregistry())).to.equal('apps.v5.iexec.eth');
+            expect(await lookup(await iexecPoco.datasetregistry())).to.equal(
+                'datasets.v5.iexec.eth',
+            );
+            expect(await lookup(await iexecPoco.workerpoolregistry())).to.equal(
+                'workerpools.v5.iexec.eth',
+            );
+        });
+
+        it('Should register reverse resolution name', async () => {
+            const name = 'test.domain.eth';
+            const reverseNameHash = ethers.utils.namehash(
+                `${proxyAddress.substring(2)}.addr.reverse`,
+            );
+            const reverseRootNameHash = ethers.utils.namehash('addr.reverse');
+            const reverseRegistrarAddress = await ensRegistry.owner(reverseRootNameHash);
+            const reverseResolverAddress = await ReverseRegistrar__factory.connect(
+                reverseRegistrarAddress,
+                anyone,
+            ).defaultResolver();
+            const reverseResolver = PublicResolver__factory.connect(reverseResolverAddress, anyone);
+            await expect(iexecPocoAsAdmin.setName(ensRegistry.address, name))
+                .to.emit(reverseResolver, 'NameChanged')
+                .withArgs(reverseNameHash, name);
+            expect(await lookup(proxyAddress)).to.equal(name);
+        });
+
+        it('Should not register reverse resolution name when sender is not the owner', async () => {
+            await expect(
+                iexecPoco.setName(ensRegistry.address, 'some.name.eth'),
+            ).to.be.revertedWith('Ownable: caller is not the owner');
+        });
+    });
+
+    /**
+     * Get the address associated to the given ENS name using forward resolution.
+     * @param domain ENS domain name
+     * @returns ETH address
+     */
+    async function resolve(domain: string) {
+        const nameHash = ethers.utils.namehash(domain);
+        const resolver = await getResolver(nameHash);
+        return await resolver['addr(bytes32)'](nameHash);
```

Comment:
I don't get this operation

Reply:
The steps to resolve an ENS name are:
1. Call ENS registry to get the resolver associated to the name => `getResolver()`
2. Query the resolver to get the address mapped to the name => `resolver.addr()`

I recommand you check this for in-depth explanation: https://docs.ens.domains/resolution.

The form `['addr(bytes32)']` is used to specify the exact signature of the function `addr` because there are multiple declarations (`addr(bytes32 node)`, `addr(bytes32 node, uint256 coinType)`, ...)

Reply:
Thanks for the informations !